### ISSUE TITLE
Remove add_kubernetes_metadata from default Agent's configs

### ISSUE THREAD

**Issue by ChrsMark** [elastic/beats#29767]
As it is reported in various [issues](https://github.com/elastic/sdh-beats/issues/1554#issuecomment-1000775087) the automatic enablement of `add_kubernetes_metadata` processor for Filebeat can bring some trouble when Filebeat is started by Agent using the default configs without the option to configure them somehow. 

In this regard we need to evaluate removing this `add_kubernetes_metadata: ~` from the default configuration, so as to avoid hitting such issues while at the same time the automatically started processor does not serve much in k8s environments like other processors do.
 In the past we rejected this at elastic/beats#27216 cause it would be a breaking change but now I'm more confident that such change would be of benefit.

@ruflin @tetianakravchenko @MichaelKatsoulis let me know what you think about this. If we agree on removing I will proceed and open a PR to fix this.

**Comment by MichaelKatsoulis** [comment]
+1

**Comment by tetianakravchenko** [comment]
+1

**Comment by jsoriano** [comment]
+1 for doing this, but I would say that we have to slip this into 8.0, for the cases where it can be a breaking change.

The main breaking case I can think of is Filebeat collecting logs using an input, but in this case the user has already modified the config file, and wouldn't break on update. We could explicitly document this use case somewhere (if it is not already documented), and remark there the need of `add_kubernetes_metadata` for enrichment.

**Comment by ruflin** [comment]
The linked discussion goes into removing it as default when run by Elastic Agent or Beats. What is the proposal here? 

It's pretty late for 8.0. Is it enabled in the config by default or also in the code? Because if we have it in the config as default and would remove it, someone upgrading beats would not be affected by it because the config stays during upgrade. @jsoriano Is this also the scenario you are hinting at?

**Comment by ChrsMark** [comment]
As it is mentioned in the description of this issue the proposal is to completely remove the `add_kubernetes_metadata: ~` configuration from the default Beat's config files. For example from https://github.com/elastic/beats/blob/eef391b64814efc379405653a751b76d4b3a7ea5/filebeat/filebeat.yml#L165.


As it is stated at #27216 and the other issues, we hit this issue in Agent when it starts Filebeat with the default configuration `filebeat.yml`. Based on discussion took place at  #27216 we don't have a way to change the underlying Beat's configurations when they are started by Agent. In this regard removing the `add_kubernetes_metadata: ~`  enablement from Beat's default configs would solve the issue in Agent. 

In Beat's side I would consider it more a cleanup since from my experience with the processor I find it quite rare that it would work with the default config `~` and most of the times the processor's config requires some mangling so as to make it work though the proper matchers etc. 

As it is clear the processor is not enabled by default in the code but by adding the `add_kubernetes_metadata: ~` setting in the default configuration that we ship.

**Comment by ruflin** [comment]
Thanks for the details @ChrsMark It means the cleanup happens in Beats but that means also Elastic Agent. 

I'm onboard with making the change and I think I now better understand the benefits/problems. My biggest concern is that we might break some user deployments. As we are late in the 8.0 cycle, I would prefer we make the change in 8.1 and not breaking or have a migration path.

**Comment by ChrsMark** [comment]
I have came up with a workaround at https://github.com/elastic/beats/pull/29809. I added a condition so as to skip the processor when the events are produced by Beats controlled by Elastic Agent. I achieve this be adding a condition to check `when.not.has_fields: ['elastic_agent']`. 

I tested manually the case with this addition and I cannot reproduce the original issue. If we agree on this we can backport it as a bugfix to 8.0 and even to 7.17.
